api/src/controllers/record-controller.js

```diff
@@ -433,4 +489,34 @@ let addStandardQueryFilters = function(query, args) {
 
   return query;
 };
+
+let processPostRequest = async function (args, res, next, property, data) {
+  var i = data.length - 1;
+  var observables = [];
+
+  do {
```

Comment:
With the do-while loop, we probably need to check that 'i' is > 0.  Otherwise your first loop would try and access data[-1]